### PR TITLE
Translate products inside productcatalogservice (at runtime)

### PR DESCRIPTION
**Pull request by NimJay**
* This is not meant to be merged into `main`.
* The changes from this pull-request include:
    * translation of productcatalogservice products (via runtime requests to Google Translate API)
    * translation of frontend strings
    * overriding of incorrect Google Translate API results
    * a new Kustomize component, `https-frontend-domain`, for configuring an HTTPS custom domain

### CODE REVIEW

File: protos/demo_grpc.pb.go
Comment (NimJay) [inline protos/demo_grpc.pb.go:1]: To regenerate the gRPC code, I ran:
```
protoc --go_out=. --go_opt=paths=source_relative \
    --go-grpc_out=. --go-grpc_opt=paths=source_relative \
    Demo.proto
```
This generated two files:
```
demo.pb.go
demo_grpc.pb.go
```
Previously, we only had 1 file (`demo.pb.go`).

Comment (NimJay) [reply protos/demo_grpc.pb.go:1]: I had to install `protoc-gen-go` and `protoc-gen-go-grpc`, which can be installed using:
```
brew install protoc-gen-go
brew install protoc-gen-go-grpc
```
or 
```
go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.28
go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.2
export PATH="$PATH:$(go env GOPATH)/bin"
```

File: src/productcatalogservice/product_catalog.go [hunk 1]
```diff
@@ -27,6 +28,7 @@ import (
 
 type productCatalog struct {
 	catalog pb.ListProductsResponse
+	pb.UnimplementedProductCatalogServiceServer
```
Comment (NimJay) [inline src/productcatalogservice/product_catalog.go:31]: Context for adding this `pb.UnimplementedProductCatalogServiceServer` line: https://github.com/grpc/grpc-go/issues/3794#issuecomment-720599532.
**tl;dr**: It's a gRPC best practice, while a [lot of developers don't necessarily agree with](https://github.com/grpc/grpc-go/issues/3669).

**Comment by NimJay** [comment]
Closing this since Cloud Next 23 is now done.
